pages/resources/tutorials/query-history.md

```diff
@@ -0,0 +1,189 @@
+
+- Install and run Trin
+- Learn how to retrieve data from your node
+- Learn how to decode Portal Network data
+
+## Prerequisites
+
+To retrieve data from the Portal network, you need to run a Portal client. In this tutorial you will use Trin. Trin is written in Rust, so your computer will need to have Rust installed.
+
+There are several options for how you send requests to your node. In this guide you will learn how to use HTTP to send requests. The requests are formatted as JSON data, and you will send them to your node using a tool called `curl`.
+
+The responses arrive at your node encoded, so you also have to decode them to make the data human-readable. You can use many different tools for this, in this tutorial we will use a simple Rust script. 
+
+To follow this guide you need to have the following installed on your machine:
+
+- [Rust](https://www.rust-lang.org/tools/install)
+- [cURL](https://curl.se/)
+- [Trin](https://github.com/ethereum/trin)
+
+There are specific instructions for installing Trin in the next section.
+
+
+## Install Trin
+
+Trin turns your computer into a Portal Network node, meaning it can relay requests to other nodes on the network. Your request is passed around the network until it reaches a node storing the specific information you need. Then, that node sends the data to you.
+
+The easiest way to get Trin is to clone the [Github repository](https://github.com/ethereum/trin).
+
+With Trin downloaded, you can build the executable by running the following command from the project root:
+
+`cargo build --workspace --release`
+
+This will create a `trin` binary in `~/trin/target/release/trin`. You can run trin from there, or move the binary into a more convenient location. In this tutorial it is assumed you moved your Trin binary into `usr/local/bin`, as follows:
+
+```sh
+cp -a ~/trin/target/release/trin /usr/local/bin/trin
+```
+You can now run Trin! 
+
+However, for this tutorial you will need to add some flags to ensure Trin exposes an HTTP port to receive your requests. You can do this with `--web3-transport http`. Your command to start Trin is therefore:
+
+```sh
+/usr/local/bin/trin --web3-transport http
+```
+
+By default, Trin exposes `http://localhost:8545` for HTTP traffic.
+
+You can leave Trin running for the remainder of this tutorial.
+
+
+## Making a request
+
+The Portal network has its own set of [RPC methods](https://github.com/ethereum/portal-network-specs/blob/master/jsonrpc/README.md) that are used to look up specific data on the network. 
+
+In this guide you will use the `portal_historyRecursiveFindContent` method. This is for looking up a specific piece of content on the Portal history network.
+
+To define which specific piece of data you want, you have to construct a **content key**. A [content key](https://github.com/ethereum/portal-network-specs/blob/master/content-keys-test-vectors.md) is a string of hex-encoded bytes made up from an identifier and a block hash.
```

Comment:
This might work well as it's own tutorial.  A dedicated informational resource on how portal addresses content and explaining the content key/content-id system seems like a nice thing to be able to link to and have available.

Reply:
agree - coming next!
When I;ve created a dedicated page for that, I'll come back here and de-duplicate.